pandas/io/pytables.py

```diff
@@ -4124,7 +4124,7 @@ def process_filter(field, filt):
                             # the levels
                             if self.is_multi_index:
                                 filt = filt.union(Index(self.levels))
-
+                            # pylint: disable-next=cell-var-from-loop
```

Comment:
I think this is a good one to fix, as the code is really nested here...can you factor out `process_filter` and define it on L4108 (before the if-statement), and give it an extra argument `op`, which you can then pass in on L4128

Reply:
Made change, waiting to see what `pre-commit run --all-files` tells me before going for `git push`.